regparse.c

```diff
@@ -5213,6 +5213,7 @@ parse_enclose(Node** np, OnigToken* tok, int term, UChar** src, UChar* end,
 	  case '-':  neg = 1; break;
 	  case 'x':  ONOFF(option, ONIG_OPTION_EXTEND,     neg); break;
 	  case 'i':  ONOFF(option, ONIG_OPTION_IGNORECASE, neg); break;
+    case 'f':  ONOFF(option, ONIG_OPTION_NOMATCHDATA, neg); break;
```

Comment:
This means you can write 

```ruby
/(foo) (?-f:bar) (baz)/.match("foo bar baz") # do not capure bar
```

... which I think is overly complicated.  Do we need to avoid allocating _a part_ of a regular expression?

Reply:
Yeah, this construction does not make sense. This option was meant to use for the whole regex, like `/(foo) (?:bar) (baz)f/.match("foo bar baz")`. So in this sense it should behave more like `/o` option, than like `/i` or `/m`.

I investigated how `\o` and `\i` \ `\m` \ etc are implemented, but can't figure out how to implement this option with those new requirements. Do you have any hints?

Reply:
@fatkodima `/.../o` is handled here:

https://github.com/ruby/ruby/blob/445e5548c9da906a2d7a490e660328b2893d07d1/parse.y#L6869-L6871

This one happens during scanning a regex terminator (that is, "`/o`" part is a lexter token whose semantic value is `RE_OPTION_ONCE`).  The terminal symbol is then passed to `new_regexp()` in the same file, where it calls `regex_compile()`.   I think you want to look that part around.

Reply:
Updated with suggestion. This is now ready for review.